gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/util/CoreTestHelper.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.tinkerpop.gremlin.util;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.tinkerpop.gremlin.structure.io.Storage;
+
+/**
```

Comment:
I've been hesitant to move a "test" class to gremlin-core because such classes have no use outside of TinkerPop test cases. I preferred the minor code duplication that we had before. I'd say that I'm +0 for this change, however if you feel strongly that the duplication should be removed, please include some javadoc here on this class to explain that this class is really for testing purposes only and that its API may change without notice. That way if someone uses it (you'd be surprised at that classes we've made public before that people find a way to use) we absolve ourselves of some of the responsibility for maintaining the API. I can't think of any other way to restrict this class specifically to "test" except to add a new `gremlin-test-core`  module or something, but I can't think of any other code that would be helpful in such a module, so that seems like overkill.

Reply:
The problem was that there were two TestHelper classes with exactly the same qualified name with different contents that confused the IDE. In addition, the changes I introduced for unification of the test directories. Therefore I extracted the unified functionality in a new CoreTestHelper class, so that the second TestHelper could inherit from CoreHestHelper and thus reduced the duplication to a single method in a single class.

Reply:
Thanks for the explanation. I've never had the IDE problems, but as I said, i'm indifferent to the change (+0 meaning i'm not for it, but won't stand in the way of it merging as it is). I'd only ask for some javadoc to explain that this class is really for testing purposes only and that its API may change without notice. I can add it myself on merge if you're unable to get to doing that.